ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/TransactionContractKeysValidation.scala

```diff
@@ -0,0 +1,295 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.participant.state.kvutils.committer.transaction
+
+import com.daml.ledger.participant.state.kvutils.Conversions
+import com.daml.ledger.participant.state.kvutils.DamlKvutils.{
+  DamlContractKey,
+  DamlStateKey,
+  DamlStateValue,
+}
+import com.daml.ledger.participant.state.kvutils.committer.Committer.Step
+import com.daml.ledger.participant.state.kvutils.committer.transaction.TransactionCommitter.{
+  DamlTransactionEntrySummary,
+  damlContractKey,
+  globalKey,
+}
+import com.daml.ledger.participant.state.kvutils.committer.{StepContinue, StepResult}
+import com.daml.ledger.participant.state.v1.RejectionReason
+import com.daml.lf.data.Ref.TypeConName
+import com.daml.lf.data.Time.Timestamp
+import com.daml.lf.transaction.{Node, NodeId}
+import com.daml.lf.value.Value
+import com.daml.lf.value.Value.ContractId
+
+object TransactionContractKeysValidation {
+
+  private type RawContractId = String
+
+  private[transaction] def validate(
+      transactionCommitter: TransactionCommitter
+  ): Step[DamlTransactionEntrySummary] =
+    (commitContext, transactionEntry) => {
+      val damlState = commitContext
+        .collectInputs[(DamlStateKey, DamlStateValue), Map[DamlStateKey, DamlStateValue]] {
+          case (key, Some(value)) if key.hasContractKey => key -> value
+        }
```

Comment:
We used to append `commitContext.getOutputs` to this. Is there a reason we don't do that any more?

Reply:
Right, I forgot to add a note about that: we were adding them before they were populated and it also seemed logically wrong to me, as the keys validation is supposed to work on the initial daml state.

Reply:
Right, makes sense.

Reply:
Can you prove that it's wrong? As in, can you write a test case that fails with the old code?

Reply:
Adding outputs at this validation stage didn't really do anything anyway, as they are still empty in the context and set later, at the `blind` stage. The previous code was just convoluted and logically misleading, but it did the exact same thing.